nncf/experimental/torch/fx/transformations.py

```diff
@@ -174,11 +207,16 @@ def constant_update_fn(model: torch.fx.GraphModule, node: torch.fx.Node, value:
 
     # Update metadata of the new constant node.
     previous_const = args[input_port_id]
-    new_constant.meta = copy(previous_const.meta)
-    new_constant.meta["val"] = value
+    consumer_nodes = list(
+        previous_const.users.keys()
+    )  # This list of consumer nodes will always be topologically sorted
+    # To ensure the updated node has the right order,
+    # we insert constant node before the node placed at the highest order in topological order.
+    with graph.inserting_before(consumer_nodes[0]):
```

Comment:
1) Why should we do this? Please describe it in the comment
2) Please add a unit test, use this `test_constant_update_transformation` from https://github.com/openvinotoolkit/nncf/pull/2920 as a basis

Reply:
We need this to ensure that the new constant node is inserted before the node which appears first topologically in the graph. For eg:
    const
    /     \
conv     \
   |         /
matmul
in this case, if we call update constant from matmul, the earlier code would have inserted it before matmul in the graph and since conv appears first in the graph, the topology is disrupted which causes errors. Hence, the code first checks all the consumer nodes of the constant node and then inserts the new constant before the node which appears first(conv node here)

I have written a unit test for this case already, I will checkout the test mentioned in the PR as well!